rwengine/src/script/modules/GTA3ModuleImpl.inl

```diff
@@ -2556,9 +2576,16 @@ void opcode_00c6(const ScriptArguments& args) {
 	@arg arg1 
 */
 void opcode_00d6(const ScriptArguments& args, const ScriptInt arg1) {
-	RW_UNIMPLEMENTED_OPCODE(0x00d6);
-	RW_UNUSED(arg1);
-	RW_UNUSED(args);
+	if (arg1 <= 7) {
+		args.getThread()->conditionCount = arg1+1;
+		args.getThread()->conditionMask = 0xFF;
+		args.getThread()->conditionAND = true;
+	}
+	else {
+		args.getThread()->conditionCount = arg1-19;
```

Comment:
beyond the scope:

This probably needs some documentation? Why -19?
This makes it possible to have a negative condition count if 8 <= arg1 <= 18
Are these bitmasks by any chance?

(The `arg1 <= 7` is not very clear to begin with either)